hera_cal/delay_filter.py

```diff
@@ -194,19 +198,27 @@ def load_delay_filter_and_write(datafile_list, baseline_list=None, calfile_list=
             if factorize_flags:
                 logger.info("  Factorizing flags")
                 df.factorize_flags(time_thresh=time_thresh, inplace=True)
+            if apply_flag_to_nsample:
+                logger.info("  Applying flags to nsample arrays")
+                for bl in baseline_list[i:i + Nbls_per_load]:
+                    _nsample = np.copy(df.hd.get_nsamples(bl))
+                    _nsample[df.hd.get_flags(bl)] = 0
+                    if _nsample.ndim == 2:
+                        _nsample = _nsample[:, :, np.newaxis]
+                    df.hd.set_nsamples(_nsample, bl)
```

Comment:
This modifies self.hd but not self.nsamples (see https://github.com/HERA-Team/hera_cal/blob/d45df271d6fc16519e22b2a78c89c2cf92e2b367/hera_cal/vis_clean.py#L481)

Are you sure that's what you want to do?

(Unrelated, this is one of my big issues with vis_clean and its related classes... it attaches data is multiple ways that are had to keep track of. It's why I've generally been avoiding vis_clean and related classes for H6C and later.)

Reply:
That's a very good question, thanks for raising this. 

It seems to be more complicated than I thought. Here, when writing to file we set `partial_write=Nbls_per_load < len(baseline_list)`. If `partial_write` is `True`, we write from `self.hd`, otherwise we construct a new data container using `self.nsamples`. I am not exactly sure the reasoning behind these two modes of writing, maybe I can add a condition so that I modify `self.hd.nsamples` vs `self.nsamples` depending on the `partial_write` value we will be using in the end?

Reply:
Maybe modify both?